code/modules/research/xenobiology/xenobiology.dm

```diff
@@ -231,9 +256,23 @@
 		qdel(src)
 	else
 		to_chat(user, "<span class='notice'>[M] looks interested for a moment, but then looks back down. Maybe you should try again later.</span>")
+		heat_stage += 1
+		heat_cooldown()
+		if(user.Adjacent(src))
+			switch(heat_stage)
+				if(1)
+					to_chat(user, "<span class='warning'>An intense heat emanates from [src]. It might need to cool off for awhile.</span>")
+				if(2)
+					to_chat(user, "<span class='warning'>[src] is boiling hot! You shudder to think what would happen if you used it again...</span>")
 		being_used = FALSE
 		..()
 
+/obj/item/slimepotion/sentience/proc/heat_cooldown()
+	if(!heat_stage)
+		return
+
+	addtimer(VARSET_CALLBACK(src, heat_stage, (heat_stage - 1)), 60 SECONDS)
```

Comment:
if this is called twice in less than 60 seconds, is there a possibility that it will never return to the initial heat_stage value?

Reply:
Yup, fixed this by making the timer a proc callback instead of varset and splitting some behavior into another proc. Works as expected now.